File: common/src/main/java/com/zorroa/common/elastic/SearchHitRowMapper.java
package com.zorroa.common.elastic;


import org.elasticsearch.search.SearchHit;

/**
 * An interface used by ElasticTemplate for mapping documents from an Elastic search result
 * on a per-document basis.  Implementations of this interface perform the actual work of
 * mapping each document to an object.
 *
 * @author chambers
 *
 * @param <T>
 */
public interface SearchHitRowMapper<T> {

     T mapRow(SearchHit hit) throws Exception;

}

File: archivist/src/main/kotlin/com/zorroa/archivist/elastic/SearchHitRowMapper.kt
package com.zorroa.archivist.elastic


/**
 * An interface used by ElasticTemplate for mapping documents from an Elastic search result
 * on a per-document basis.  Implementations of this interface perform the actual work of
 * mapping each document to an object.
 *
 * @author chambers
 *
 * @param <T>
</T> */
interface SearchHitRowMapper<T> {

    @Throws(Exception::class)
    fun mapRow(hit: SingleHit): T

}
